apps/src/dance/ai/DanceAiClient.tsx

```diff
@@ -0,0 +1,107 @@
+import UntypedCachedBackgrounds from '@cdo/static/dance/ai/model/cached-spacy-background-map.json';
+import UntypedCachedForegrounds from '@cdo/static/dance/ai/model/cached-spacy-foreground-map.json';
+import UntypedCachedPalettes from '@cdo/static/dance/ai/model/cached-spacy-palette-map.json';
+
+type CachedWeightsMapping = {
+  emojiAssociations: {[key: string]: number[]};
```

Comment:
not blocking, but if we do have a list of emojis and list of effect types (either in a JSON file or as a typescript type), you could further narrow this type so that the keys have to be one of the emojis (something like `emojiAssociations: {[key in Emoji]: number[]}; output: Output[];`)

Reply:
that makes sense, we're currently still evaluating what the final list of emojis will be but we can definitely implement that type check once it is finalized. In addition, we anticipate adding song metadata as a form of input so the json structure will inevitably change as a result of that as well.